File: functions/updated_financial_calculator.py
# Lila Shearer, Financial Calculator Update - Python

print("\nWelcome to Financial Calculator!\nThis program is designed to calculate what percentage of your income that expenses and savings take up, and show how much you have left to spend.\n")
print("When answering questions, please write numbers and do not include the dollar sign (example: 256.78). Thanks!\n")  

income = float(input("What is your income?\n"))

def info(cost, income, type):
    percent = (cost/income * 100)
    print(f"That amount of money that goes to {type} is ${cost:.2f}, which is {percent:.1f}% of your monthly income.")

def userInfo(expense):
    return float(input(f"How much do you spend on {expense}?\n"))

rent = userInfo("rent")
utilities = userInfo("utilities")
groceries = userInfo("groceries")
transportation = userInfo("transportation")

savings = income/10
spending = income-rent-utilities-groceries-transportation-savings

print("\nHere are your results!\n")

info(rent, income, "rent")
info(utilities, income, "utilities")
info(groceries, income, "groceries")
info(transportation, income, "transportation")
info(savings, income, "savings")
info(spending, income, "spending")

print("\nThank you for using Financial Calculator!\nHave a great day!\n")

File: functions/updated_financial_calculator.c
//Lila Shearer, Updated Financial Calculator - C
#include <stdio.h>
#include <math.h>

void info(float cost, float income, char type[50]){
    float percent = cost/income*100;
    printf("The amount of money that goes to %s is $%.2f, which is %.1f%% of your income.\n", type, cost, percent);
}

float user_info(char expense[20]){
    float answer;
    printf("How much money do you spend on %s?\n", expense);
    scanf("%f",&answer);
    return answer;
}
float income;
int main(void){
    printf("\nWelcome to Financial Calculator!\nThis program is designed to calculate what percentage of your income that expenses and savings take up, and show how much you have left to spend.\n");
    printf("\nWhen answering questions, please write numbers and do not include the dollar sign (example: 256.78). Thanks!\n");
    printf("\nWhat is your income?\n");
    scanf("%f",&income);
    float savings = income/10;
    float rent = user_info("rent");
    float utilities = user_info("utilities");
    float groceries = user_info("groceries");
    float transportation = user_info("transportation");
    float spending = (income-rent-utilities-groceries-transportation-savings);
    printf("\nHere are your results!\n");
    info(rent, income, "rent");
    info(utilities, income, "utilities");
    info(groceries, income, "groceries");
    info(transportation, income, "transportation");
    info(savings, income, "savings");
    info(spending, income, "spending");
    printf("\nThank you for using Financial Calculator!\nHave a great day!\n");
    return 0;
}